src/templates/finops-hub/modules/dataFactory.bicep

```diff
@@ -4863,6 +4939,401 @@ resource pipeline_RerunETL 'Microsoft.DataFactory/factories/pipelines@2018-06-01
+          secureInput: false
+        }
+        userProperties: []
+        typeProperties: {
+          variableName: 'pipelineFailed'
+          value: {
+            value: '@bool(false)'
+            type: 'Expression'
+          }
+        }
+      }
+      { // Load Schema Mappings
+        name: 'Load Schema Mappings'
+        type: 'Lookup'
+        dependsOn: [
+          {
+            activity: 'Set Schema Filename'
+            dependencyConditions: ['Succeeded']
+          }
+        ]
+        policy: {
+          timeout: '0.12:00:00'
+          retry: 0
+          retryIntervalInSeconds: 30
+          secureOutput: false
+          secureInput: false
+        }
+        userProperties: []
+        typeProperties: {
+          source: {
+            type: 'JsonSource'
+            storeSettings: {
+              type: 'AzureBlobFSReadSettings'
+              recursive: true
+              enablePartitionDiscovery: false
+            }
+            formatSettings: {
+              type: 'JsonReadSettings'
+            }
+          }
+          dataset: {
+            referenceName: dataset_config.name
+            type: 'DatasetReference'
+            parameters: {
+              fileName: {
+                value: '@variables(\'schemaFile\')'
+                type: 'Expression'
+              }
+              folderPath: '${configContainerName}/schemas'
+            }
+          }
+        }
+      }
+      { // Error: SchemaLoadFailed
+        name: 'Failed to Load Schema'
+        type: 'Fail'
+        dependsOn: [
+          {
+            activity: 'Load Schema Mappings'
+            dependencyConditions: ['Failed']
+          }
+        ]
+        userProperties: []
+        typeProperties: {
+          message: {
+            value: '@concat(\'Unable to load the \', variables(\'schemaFile\'), \' recommendations schema file. Please confirm the schema and version are supported for FinOps hubs ingestion. Unsupported files will remain in the ingestion container.\')'
+            type: 'Expression'
+          }
+          errorCode: 'SchemaLoadFailed'
+        }
+      }
+      { // Load Queries
+        name: 'Load Queries'
+        type: 'Lookup'
+        dependsOn: [
+        ]
+        policy: {
+          timeout: '0.00:10:00'
+          retry: 0
+          retryIntervalInSeconds: 30
+          secureOutput: false
+          secureInput: false
+        }
+        userProperties: []
+        typeProperties: {
+          source: {
+            type: 'JsonSource'
+            storeSettings: {
+              type: 'AzureBlobFSReadSettings'
+              recursive: true
+              enablePartitionDiscovery: false
+            }
+            formatSettings: {
+              type: 'JsonReadSettings'
+            }
+          }
+          dataset: {
+            referenceName: dataset_config.name
+            type: 'DatasetReference'
+            parameters: {
+              fileName: 'Recommendations.json'
```

Comment:
Make this dynamic to pull all JSON files in the queries folder.

Reply:
Hubs recommendations queries are declared in a single `Recommendations.json` file. This ADF pipeline is dedicated to process Hubs recommendations only. The queries folder might have other JSON files for other purposes, for example, for declaring KQL queries that are going to feed the governance report with resource inventories. Other queries types will have their own ADF pipeline.

Reply:
My point is we should keep this more open and not force everything to have its own pipeline, which requires more code and more maintenance and likely more bugs. We don't need to build everything now, but at a minimum we need to support non-recommendation ARG queries. Nicolas is working on that and waiting for this, so let's just make it open upfront.

Reply:
I agree with your point, but making it open is more than looking for all JSON query files. We will have multiple output schemas (recommendations, inventory, etc.) and we need to make the bridge between the query file and the respective output schema. The query file will have to refer to the schema name to be used. This will require me to refactor the pipeline to make it generic and need more time to deliver a complete solution.

Reply:
@flanakin, I just committed the changes to make the pipeline query type agnostic. Due to the ADF limitations regarding activity nesting, the solution became more complex than expected and forced me to explode all ARG queries into multiple JSON files. Let me know if it can be improved further before merging.

Reply:
I realized now there are 29 more conversations to address :-D Hold on! :-D

Reply:
Solved